### ISSUE TITLE
npm search crashes with OOM error

### ISSUE THREAD

**Issue by miksago** [npm/npm#14409]
#### I'm opening this issue because:

  - [x] npm is crashing.

#### What's going wrong?

Receiving an OOM, and crashes with stacktrace, see below.

#### How can the CLI team reproduce the problem?

Run: `npm search typescript`

No npm-debug.log produced, npm crashed with OOM.

### supporting information:

 - `npm -v` prints: 3.10.9
 - `node -v` prints: v6.4.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Mac OS X 10.11.6 (15G31)
 - Network issues:
   - Geographic location where npm was run: Berlin, Germany
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

```
$ npm search typescript
npm WARN Building the local index for the first time, please be patient
npm http request GET https://registry.npmjs.org/-/all
npm http 200 https://registry.npmjs.org/-/all

<--- Last few GCs --->

   68329 ms: Scavenge 1019.6 (1434.2) -> 1019.6 (1434.2) MB, 15.3 / 0 ms (+ 0.0 ms in 1 steps since last GC) [allocation failure].
   68344 ms: Scavenge 1019.6 (1434.2) -> 1019.6 (1434.2) MB, 15.3 / 0 ms [allocation failure].
   70715 ms: Mark-sweep 1019.6 (1434.2) -> 1019.5 (1434.2) MB, 2370.7 / 0 ms (+ 1025.3 ms in 3942 steps since start of marking, biggest step 14.5 ms) [last resort gc].
   73523 ms: Mark-sweep 1019.5 (1434.2) -> 1019.5 (1434.2) MB, 2807.9 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x240fc37c9e31 <JS Object>
    1: stringify [native json.js:164] [pc=0x131eb2333d77] (this=0x240fc37c3309 <a JSON with map 0x2dc1249095e9>,z=0x1edd956c7551 <an Object with map 0x1015996a4d49>,A=0x240fc3704189 <undefined>,N=0x240fc3704189 <undefined>)
    2: arguments adaptor frame: 1->3
    3: /* anonymous */ [/Users/em/.nvm/versions/node/v6.4.0/lib/node_modules/npm/lib/cache/update-index.js:96] [pc=0x131eb289c041] ...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 4: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 5: v8::internal::IncrementalStringBuilder::Extend() [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 6: v8::internal::BasicJsonStringifier::SerializeString(v8::internal::Handle<v8::internal::String>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 7: v8::internal::BasicJsonStringifier::Result v8::internal::BasicJsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 8: v8::internal::BasicJsonStringifier::Result v8::internal::BasicJsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
 9: v8::internal::BasicJsonStringifier::Result v8::internal::BasicJsonStringifier::Serialize_<true>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
10: v8::internal::BasicJsonStringifier::Result v8::internal::BasicJsonStringifier::Serialize_<false>(v8::internal::Handle<v8::internal::Object>, bool, v8::internal::Handle<v8::internal::Object>) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
11: v8::internal::Runtime_BasicJSONStringify(int, v8::internal::Object**, v8::internal::Isolate*) [/Users/em/.nvm/versions/node/v6.4.0/bin/node]
12: 0x131eb220961b
[1]    29437 abort      npm search typescript
```

**Comment by KenanY** [comment]
Already fixed by https://github.com/npm/npm/pull/13746, which landed in `npm@4.0.0`. Thanks for the bug report.

**Comment by miksago** [comment]
Nice! I somehow missed that when I had a quick search for this issue.. sorry!
